src/Controller/GmoLinkTypePlusController.php

```diff
@@ -166,70 +261,53 @@ public function checkValueExists($val) {
    *
    * Refer : https://docs.mul-pay.jp/linkplus/payment/common.
    */
-  public function statusMapper($state) {
+  public function statusMapper($state, $order_id) {
+    // Handles the payment flow and drupal status updated
+    // based on custom status recieved from GMO. This session 
+    // Can be more generalized. Currently we are considering only 
+    // credit, cvs, payeasy, paypay payment method's statuses
     switch ($state) {
       case 'REQPROCESS':
       case 'REQSUCCESS':
         $this->defaultPaymentStatus = 'authorization';
+        $redirect = new RedirectResponse('/checkout/' . $order_id . '/complete');
+        $str = 'Order Place Request recieved successfully. Your order will be 
+        updated soon.';
+        $this->paymentSuccess = FALSE;
+        $this->isOrderCreationNeeded = TRUE;
+        return [$redirect, $str];
         break;
 
-      case 'PAYSUCCESS':
-        $this->defaultPaymentStatus = 'Completed';
+      case 'PAYSTART':
+        $this->defaultPaymentStatus = 'new';
+        $redirect = new RedirectResponse('/checkout/' . $order_id . '/review');
+        $str = "Please review the payment details. Payment has been cancelled.";
+        $this->paymentSuccess = FALSE;
+        $this->isOrderCreationNeeded = FALSE;
+        return [$redirect, $str];
         break;
 
-      case 'EXPIRED':
-      case 'INVALID':
       case 'ERROR':
-        $this->defaultPaymentStatus = 'authorization_expired';
+        $redirect = new RedirectResponse('/checkout/' . $order_id . '/review');
+        $str = "Payment has been failed. Please check the payment details.";
+        $this->defaultPaymentStatus = 'failed';
+        $this->paymentSuccess = FALSE;
+        $this->isOrderCreationNeeded = FALSE;
```

Comment:
We need order creation for this with canceled state . These error says the user have completed payment in linktype plus but while gmo processing this request  there is an issue .